avocado/core/decorators.py

```diff
@@ -65,6 +65,7 @@ def decorator(function):
             def wrapper(*args, **kwargs):
                 raise core_exceptions.TestDecoratorSkip(message)
             function = wrapper
+        function.__skip__ = True
```

Comment:
Well this should be unique enough, but how about using something even less prone to accidental clash like `__skip_this_test_due_decorator__` or so?